src/DSi_TMD.h

```diff
@@ -0,0 +1,146 @@
+/*
+    Copyright 2016-2023 melonDS team
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#ifndef DSI_TMD_H
+#define DSI_TMD_H
+
+#include "types.h"
+#include <array>
+
+namespace DSi_TMD
+{
+
+struct TitleMetadataContent {
+    /// Content ID (00,00,00,vv) ;lowercase/hex ;"0000000vv.app"
+    u8 ContentId[4];
+
+    /// Content Index (00,00)
+    u8 ContentIndex[2];
+
+    /// Content Type (00,01) ;aka DSi .app
+    u8 ContentType[2];
+
+    /// Content Size (00,00,00,00,00,19,E4,00)
+    u8 ContentSize[8];
+
+    /// Content SHA1 (on decrypted ".app" file)
+    u8 ContentSha1Hash[20];
+
+    [[nodiscard]] u32 GetVersion() const noexcept
+    {
+        return (ContentId[0] << 24) | (ContentId[1] << 16) | (ContentId[2] << 8) | ContentId[3];
+    }
+};
+
+static_assert(sizeof(TitleMetadataContent) == 36, "TitleMetadataContent is not 36 bytes!");
+
+/// Metadata for a DSiWare title.
+/// Used to install DSiWare titles to NAND.
+/// @see https://problemkaputt.de/gbatek.htm#dsisdmmcdsiwareticketsandtitlemetadata
+struct TitleMetadata
+{
+    u32 SignatureType;
+
```

Comment:
This looks pretty blown up, I think it would make more sense to only put spaces to group struct items.

Reply:
I can slim this down.

Reply:
I slimmed it down.